### PR TITLE
Documentation/eap radius ldap

### PR DESCRIPTION
**Pull request by fdurand**
# Description
Documentation for eap profile selection regex realm and ldap authentication changes

# Impacts
The documentation

# Delete branch after merge
YES

# Checklist
- [ ] Document the feature
- [ ] Add unit tests
- [ ] Add acceptance tests (TestLink)

# NEWS file entries

## Enhancements
* Documentation for eap profile selection
* Documentation for regex realm 
* Documentation for new action/condition in ldap authentication
* Moved the vlan filters example as default disabled vlan filter.

### CODE REVIEW

File: conf/vlan_filters.conf.defaults [hunk 1]
```diff
@@ -18,3 +19,69 @@ scopes = RegisteredRole
 action.0=modify_node: mac, $mac, status = unreg, autoreg = no
 role = registration
 
+[EXAMPLE_No_Internet_for_lunch]
+run_actions=enabled
+status=disabled
+description=revent a device from connecting when its role is "default", the SSID is "SECURE", the current time is between 11am and 2pm, from Monday to Friday and as a registered device
```
Comment (extrafu) [inline conf/vlan_filters.conf.defaults]: revent -> Prevent
and as a registered device -> and is a registered device

File: conf/vlan_filters.conf.defaults [hunk 2]
```diff
@@ -18,3 +19,69 @@ scopes = RegisteredRole
...
+status=disabled
+description=Trigger a security event if the SSID is OPEN and the owner is igmout
+scopes=RegisteredRole
+action.0=trigger_security_event: mac, $mac, tid, 12345, type, CUSTOM
+condition=owner.pid == "igmout" && ssid == "OPEN"
+
+[EXAMPLE_Autoregister_if_user]
+run_actions=enabled
+status=disabled
+description=autoregister the device and assign the role default to each device where the username is igmout.
```
Comment (extrafu) [inline conf/vlan_filters.conf.defaults]: autoregister -> Autoregister
the role default -> the default role
where the username -> when the username

File: conf/vlan_filters.conf.defaults [hunk 3]
```diff
@@ -18,3 +19,69 @@ scopes = RegisteredRole
...
+scopes=AutoRegister,NodeInfoForAutoReg
+role=default
+condition=username == "igmout" && ssid == "SECURE"
+
+[EXAMPLE_Autoregister_windows_devices]
+run_actions=enabled
+status=disabled
+top_op=and
+description=autoregister all Windows devices and assign them the role default
```
Comment (extrafu) [inline conf/vlan_filters.conf.defaults]: autoregister -> Autoregister
the role default -> the default role

File: conf/vlan_filters.conf.defaults [hunk 4]
```diff
@@ -18,3 +19,69 @@ scopes = RegisteredRole
...
+scopes=AutoRegister,NodeInfoForAutoReg
+role=default
+condition=contains(fingerbank_info.device_hierarchy_names, "Windows")
+
+[EXAMPLE_Reject_specific_MAC]
+run_actions=enabled
+status=disabled
+description=filter a MAC address and reject it by assigning the role REJECT
```
Comment (extrafu) [inline conf/vlan_filters.conf.defaults]: filter -> Filter
the role REJECT -> the REJECT role

File: conf/vlan_filters.conf.defaults [hunk 5]
```diff
@@ -18,3 +19,69 @@ scopes = RegisteredRole
...
+scopes=RegistrationRole,RegisteredRole
+role=REJECT
+condition=mac == "aa:bb:cc:dd:ee:ff" && ssid == "OPEN"
+
+[EXAMPLE_Detect_VOIP]
+run_actions=enabled
+status=disabled
+description=auto set Avaya and Polycom as phones by matching vendor MAC, and set to voice role
```
Comment (extrafu) [inline conf/vlan_filters.conf.defaults]: auto -> Automatically
and sent to voice role -> and assign the voice role

File: docs/PacketFence_Installation_Guide.asciidoc [hunk 6]
```diff
@@ -2440,215 +2440,106 @@ And can be defined using different criteria like:
   owner.attribute (like owner.pid)
   radius_request.attribute (like radius_request.Calling-Station-Id)
 
-Below are several common examples of VLAN filters usage.
+There is some default vlan filters defined in the configuration you can use to achieve the following goal:
```
Comment (extrafu) [inline docs/PacketFence_Installation_Guide.asciidoc]: There is some default vlan -> There are some default VLAN

File: docs/PacketFence_Installation_Guide.asciidoc [hunk 7]
```diff
@@ -2440,215 +2440,106 @@ And can be defined using different criteria like:
...
 
-.EXAMPLE: revent a device from connecting when its role is "default", the SSID is "SECURE", the current time is between 11am and 2pm, from Monday to Friday and as a registered device
-[source,ini,numbered]
-----
-  [category]
-  filter = node_info.category
-  operator = is
-  value = default
+.EXAMPLE_Reject_between_11am_2pm will revent a device from connecting when its role is "default", the SSID is "SECURE", the current time is between 11am and 2pm, from Monday to Friday and as a registered device
```
Comment (extrafu) [inline docs/PacketFence_Installation_Guide.asciidoc]: and as a -> and is a

File: docs/PacketFence_Installation_Guide.asciidoc [hunk 8]
```diff
@@ -2440,215 +2440,106 @@ And can be defined using different criteria like:
...
-  filter = time
-  operator = is
-  value = wd {Mon Tue Wed Thu Fri} hr {11am-2pm}
+.EXAMPLE_Autoregister_if_user will autoregister the device and assign the role staff to each device where the username is igmout.
 
-  [1:category&ssid&time]
-  scope = RegisteredRole
-  role = nointernet
-----
+.EXAMPLE_Autoregister_windows_devices will autoregister all Windows devices and assign them the role default.
```
Comment (extrafu) [inline docs/PacketFence_Installation_Guide.asciidoc]: the role default -> the default role

File: docs/PacketFence_Installation_Guide.asciidoc [hunk 9]
```diff
@@ -2440,215 +2440,106 @@ And can be defined using different criteria like:
...
 
-.EXAMPLE: create a security violaton if the SSID is OPEN and the owner is igmout
-[source,ini,numbered]
-----
-  [igmout]
-  filter = owner.pid
-  operator = is
-  value = igmout
+.EXAMPLE_Reject_specific_MAC will filter a MAC address and reject it by assigning the role REJECT.
```
Comment (extrafu) [inline docs/PacketFence_Installation_Guide.asciidoc]: the role REJECT -> the REJECT role

File: docs/PacketFence_Installation_Guide.asciidoc [hunk 10]
```diff
@@ -2440,215 +2440,106 @@ And can be defined using different criteria like:
...
 
-  [open]
-  filter = ssid
-  operator = is
-  value = OPEN
+.EXAMPLE_Detect_VOIP will auto set Avaya and Polycom as phones by matching vendor MAC, and set to default role
```
Comment (extrafu) [inline docs/PacketFence_Installation_Guide.asciidoc]: will auto -> will automatically

File: docs/PacketFence_Installation_Guide.asciidoc [hunk 11]
```diff
@@ -2440,215 +2440,106 @@ And can be defined using different criteria like:
...
+^^^^^^^^^^^^^^^^^
 
-.EXAMPLE: autoregister all Windows devices and assign them the role default.
-[source,ini,numbered]
-----
-  [Windows_AutoReg]
-  filter = fingerbank_info.device_hierarchy_names
-  operator = includes
-  value = Windows
+ldapfilter action override the internal ldap filter packetfence create internally (uid=$username) so you can create a custom filter that match your needs.
```
Comment (extrafu) [inline docs/PacketFence_Installation_Guide.asciidoc]: override -> overrides
ldap -> LDAP
packetfence create -> that PacketFence creates
that match -> that matches

File: docs/PacketFence_Installation_Guide.asciidoc [hunk 12]
```diff
@@ -2440,215 +2440,106 @@ And can be defined using different criteria like:
...
 
-  [AUTOREG_WINDOWS:Windows_AutoReg]
-  scope = AutoRegister
-  role = default
+Per example something like this (search for the user and check to see if it's permitted based on some criteria):
```
Comment (extrafu) [inline docs/PacketFence_Installation_Guide.asciidoc]: Per example -> For example

File: docs/PacketFence_Installation_Guide.asciidoc [hunk 13]
```diff
@@ -2440,215 +2440,106 @@ And can be defined using different criteria like:
...
-  operator = match
-  value = aa:bb:cc:dd:ee:ff
+Set_role_on_not_found action
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^
 
-  [SSID_Open]
-  filter = ssid
-  operator = is
-  value = Open
+set_role_on_not_found is a way to define a role is the rule doesn't match, let's take the ldapfilter example above.
```
Comment (extrafu) [inline docs/PacketFence_Installation_Guide.asciidoc]: is the rule -> if the rule

File: docs/PacketFence_Installation_Guide.asciidoc [hunk 14]
```diff
@@ -2440,215 +2440,106 @@ And can be defined using different criteria like:
...
 
-  [REJECT:MAC&SSID_Open]
-  scope = RegistrationRole
-  role = REJECT
-----
+If we add the action set_role_on_not_found = REJECT so it mean that the device will be rejected if the ldap filter doesn't return anything. (if it match then set_role action will be applied)
```
Comment (extrafu) [inline docs/PacketFence_Installation_Guide.asciidoc]: so it mean -> so it means
ldap filter -> LDAP filter
if it match -> if it matches

File: docs/PacketFence_Installation_Guide.asciidoc [hunk 15]
```diff
@@ -2440,215 +2440,106 @@ And can be defined using different criteria like:
...
-  operator = is_not
-  value = pending
+role_from_source action
+^^^^^^^^^^^^^^^^^^^^^^^
 
-  [wireless_dot1x]
-  filter = connection_type
-  operator = is
-  value = Wireless-802.11-EAP
+role_from_source will check to see if the ldap attribute exist and will add it in the ldap_attribute context (available in the radius filters)
```
Comment (extrafu) [inline docs/PacketFence_Installation_Guide.asciidoc]: ldap attribute exist -> LDAP attribute exists

File: docs/PacketFence_Installation_Guide.asciidoc [hunk 16]
```diff
@@ -2440,215 +2440,106 @@ And can be defined using different criteria like:
...
 
-  [wired_dot1x]
-  filter = connection_type
-  operator = is
-  value = Ethernet-EAP
+So per example if you want to take the value of the ldap attribute customRadius and add it in the radius answer you needs to do the following.
```
Comment (extrafu) [inline docs/PacketFence_Installation_Guide.asciidoc]: So per example -> So for example
ldap attribute -> LDAP attribute
radius -> RADIUS
you needs -> you need

File: docs/PacketFence_Installation_Guide.asciidoc [hunk 17]
```diff
@@ -2440,215 +2440,106 @@ And can be defined using different criteria like:
...
 
-  [autoreg_wireless_dot1x:wireless_dot1x&not_pending]
-  scope = AutoRegister
-  role = default
+In the authentication rule, set an action "Role from source" to customRadius.
+Next create a radius filter that will add the custom radius attributes:
```
Comment (extrafu) [inline docs/PacketFence_Installation_Guide.asciidoc]: radius -> RADIUS  -  both places

File: docs/PacketFence_Installation_Guide.asciidoc [hunk 18]
```diff
@@ -2440,215 +2440,106 @@ And can be defined using different criteria like:
...
 
-  [autoreg_wired_dot1x:wired_dot1x&not_pending]
-  scope = AutoRegister
-  role = default
-----
+  [IF_SET_ROLE_FROM_SOURCE]
+  status=enabled
+  answer.0=reply:Packetfence-Raw = $ldap_attribute.customRadius
+  top_op=and
+  description=If the role has been computed from the action set_role_from_source then return the value of the role as a radius attribute
```
Comment (extrafu) [inline docs/PacketFence_Installation_Guide.asciidoc]: radius -> RADIUS

File: docs/PacketFence_Installation_Guide.asciidoc [hunk 19]
```diff
@@ -2440,215 +2440,106 @@ And can be defined using different criteria like:
...
+  condition=action == "set_role_from_source"
 
-.EXAMPLE: auto set Avaya and Polycom as phones by matching vendor MAC, and set to default role
-[source,ini,numbered]
-----
-  [Polycom]
-  filter = node_info.mac
-  operator = regex
-  value = ^(00:04.f2).*
+Note, this support multiples ldap attributes, like customRadius:Airespace-Interface-Name=internet , customRadius:Aruba-User-Vlan=666
```
Comment (extrafu) [inline docs/PacketFence_Installation_Guide.asciidoc]: this support -> this supports
ldap -> LDAP

File: docs/PacketFence_Installation_Guide.asciidoc [hunk 20]
```diff
@@ -2440,215 +2440,106 @@ And can be defined using different criteria like:
...
 
-  [avaya]
-  filter = node_info.mac
-  operator = regex
-  value = ^(00:04:0d|00:1b:4f|00:e0:07|04:8a:15|24:d9:21|2c:f4:c5|34:75:c7|3c:3a:73|3c:b1:5b|44:32:2a|58:16:26|6c:fa:58|70:30:18|70:38:ee|70:52:c5|80:1d:aa|84:83:71|90:fb:5b|a0:51:c6|b4:b0:17|c8:f4:06|cc:f9:54|d4:ea:0e|fc:83:99|fc:a8:41|b0:ad:aa|10:cd:ae|50:cd:22|b4:47:5e|d4:78:56|c0:57:bc|38:bb:3c|e4:5d:52|a4:25:1b|6c:a8:49|a0:12:90|f8:15:47|50:61:84|bc:ad:ab|b4:a9:5a|f8:73:a2|64:c3:54|64:a7:dd|64:6a:52|00:09:6e|00:07:3b|00:0d:18|00:0d:28|00:04:2e).*
+Append search attributes ldap filter
```
Comment (extrafu) [inline docs/PacketFence_Installation_Guide.asciidoc]: ldap -> LDAP

File: docs/PacketFence_Installation_Guide.asciidoc [hunk 21]
```diff
@@ -2440,215 +2440,106 @@ And can be defined using different criteria like:
...
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
 
-  [set_voip:(avaya|Polycom)]
-  scope = IsPhone
-  role = default
-----
+This option will add as a & condition to the ldap filter generated by PacketFence,
```
Comment (extrafu) [inline docs/PacketFence_Installation_Guide.asciidoc]: ldap -> LDAP

File: docs/PacketFence_Installation_Guide.asciidoc [hunk 22]
```diff
@@ -2440,215 +2440,106 @@ And can be defined using different criteria like:
...
+So per example the ldapfilter generated by PacketFence will be the following:
```
Comment (extrafu) [inline docs/PacketFence_Installation_Guide.asciidoc]: per example -> for example

File: docs/PacketFence_Installation_Guide.asciidoc [hunk 23]
```diff
@@ -2440,215 +2440,106 @@ And can be defined using different criteria like:
...
-  operator = is
-  value = Ethernet-EAP
+  (&(|(sAMAccountName=%{User-Name})(sAMAccountName=%{Stripped-User-Name})(cn=%{User-Name})(cn=%{Stripped-User-Name})(sAMAccountName=%{%{Stripped-User-Name}:-%{User-Name}})))
 
-  [EAPTLS]
-  filter = radius_request
-  attribute = EAP-Type
-  operator = is
-  value = TLS
+and if you define a appended ldap filter like:
```
Comment (extrafu) [inline docs/PacketFence_Installation_Guide.asciidoc]: ldap -> LDAP

File: docs/PacketFence_Installation_Guide.asciidoc [hunk 24]
```diff
@@ -2440,215 +2440,106 @@ And can be defined using different criteria like:
...
+  (&(|(sAMAccountName=%{User-Name})(sAMAccountName=%{Stripped-User-Name})(cn=%{User-Name})(cn=%{Stripped-User-Name})(sAMAccountName=%{%{Stripped-User-Name}:-%{User-Name}}))(|(memberOf=CN=Staff,OU=Security Groups,DC=ad,DC=acme,DC=com)(wifi=enabled)))
 
-.EXAMPLE: autoregister printers and scanners
-[source,ini,numbered]
-----
-  [printers]
-  filter = node_info.device_class
-  operator = is
-  value = Printers/Scanners
+By doing that, event you wont don't need the "Search Attributes" feature, you will be able to store the user dn in the PacketFence-UserDN attribute.
```
Comment (extrafu) [inline docs/PacketFence_Installation_Guide.asciidoc]: event you -> even if you
user dn -> user's DN

File: docs/PacketFence_Installation_Guide.asciidoc [hunk 25]
```diff
@@ -2440,215 +2440,106 @@ And can be defined using different criteria like:
...
-  scope = AutoRegister
-  role = printers
+basedn condition
+^^^^^^^^^^^^^^^^
 
-  [printers_role:(printers)]
-  scope = NodeInfoForAutoReg
-  role = printers
-----
+This condition allow to override the default basedn in the ldap source and it will permit to test if a object is in a specific ou.
```
Comment (extrafu) [inline docs/PacketFence_Installation_Guide.asciidoc]: ldap -> LDAP

File: docs/PacketFence_Installation_Guide.asciidoc [hunk 26]
```diff
@@ -2440,215 +2440,106 @@ And can be defined using different criteria like:
...
-  [Add_Note_HP_Printer:HP_Printer]
-  scope = RegisteredRole
-  role = printers
-  action = modify_node
-  action_param = mac = $mac, notes = HP PRINTER
-----
+Advanced Realm Configuration
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+In PacketFence you can define multiple Realms to select on which domain you want to authenticate the users.
```
Comment (extrafu) [inline docs/PacketFence_Installation_Guide.asciidoc]: Realms -> realms

File: docs/PacketFence_Installation_Guide.asciidoc [hunk 27]
```diff
@@ -2440,215 +2440,106 @@ And can be defined using different criteria like:
...
+
+You can define a Realm with a regex in order to match multiples format.
```
Comment (extrafu) [inline docs/PacketFence_Installation_Guide.asciidoc]: Realm -> realm
multiples format -> multiple formats.

File: docs/PacketFence_Installation_Guide.asciidoc [hunk 28]
```diff
@@ -2440,215 +2440,106 @@ And can be defined using different criteria like:
...
+
+Per example in the ACME realm we define the regex like this:
```
Comment (extrafu) [inline docs/PacketFence_Installation_Guide.asciidoc]: Per example -> For example

File: docs/PacketFence_Installation_Guide.asciidoc [hunk 29]
```diff
@@ -2440,215 +2440,106 @@ And can be defined using different criteria like:
...
+
+  .*\.acme\.com$
+
+It mean that if you have a user coming with this username mickey@la.acme.com , FreeRADIUS will define the realm as la.acme.com (it will be included in the radius request) and PacketFence will map the user to the ACME Realm.
```
Comment (extrafu) [inline docs/PacketFence_Installation_Guide.asciidoc]: It mean -> It means
FreeRADIUS? I think we should always say PacketFence - they don't know/care we're using FreeRADIUS.

Comment (extrafu) [reply docs/PacketFence_Installation_Guide.asciidoc]: Realm -> realm

File: docs/PacketFence_Installation_Guide.asciidoc [hunk 30]
```diff
@@ -2662,6 +2553,7 @@ These rules are available in thoses scopes:
   returnAuthorizeRead is when you return the answer for the switch read login access
   returnAuthorizeWrite is when you return the answer for the switch write login access
   returnAuthorizeVoip is when you return the answer for a VoIP
+  preProcess is when you want to manipulate the radius context (like adding custom attributes to the request)
```
Comment (extrafu) [inline docs/PacketFence_Installation_Guide.asciidoc]: radius -> RADIUS

File: docs/PacketFence_Installation_Guide.asciidoc [hunk 31]
```diff
@@ -2670,8 +2562,12 @@ These rules are available in thoses scopes:
   packetfence-tunnel.authorize call the radius filter in the packetfence-tunnel authorize section
   packetfence.preacct call the radius filter in the packetfence preacct section
   packetfence.accounting call the radius filter in the packetfence accounting section
+  eduroam.authorize call the radius filter in the eduroam accounting section
```
Comment (extrafu) [inline docs/PacketFence_Installation_Guide.asciidoc]: radius -> RADIUS

File: docs/PacketFence_Installation_Guide.asciidoc [hunk 32]
```diff
@@ -2670,8 +2562,12 @@ These rules are available in thoses scopes:
...
+  eduroam.pre-proxy  call the radius filter in the pre-proxy accounting section
```
Comment (extrafu) [inline docs/PacketFence_Installation_Guide.asciidoc]: radius -> RADIUS

File: docs/PacketFence_Installation_Guide.asciidoc [hunk 33]
```diff
@@ -2670,8 +2562,12 @@ These rules are available in thoses scopes:
...
+  eduroam.post-proxy call the radius filter in the post-proxy accounting section
```
Comment (extrafu) [inline docs/PacketFence_Installation_Guide.asciidoc]: radius -> RADIUS

File: docs/PacketFence_Installation_Guide.asciidoc [hunk 34]
```diff
@@ -2670,8 +2562,12 @@ These rules are available in thoses scopes:
...
+  eduroam.preacct call the radius filter in the eduroam preacct section
```
Comment (extrafu) [inline docs/PacketFence_Installation_Guide.asciidoc]: radius -> RADIUS

File: docs/PacketFence_Installation_Guide.asciidoc [hunk 35]
```diff
@@ -2693,61 +2589,66 @@ And can be defined using different criteria like:
...
-  filter = connection_type
-  operator = is
-  value = Ethernet-EAP
-
-  [1:etherneteap&!security_event]
-  merge_answer = no
-  scope = returnRadiusAccessAccept
+  [Ethernet-EAP-Accept]
+  status=enabled
+  description=Access Accept when the connection is Ethernet-EAP and when there is no security event
```
Comment (extrafu) [inline docs/PacketFence_Installation_Guide.asciidoc]: Access Accept -> Return Access-Accept

File: docs/PacketFence_Installation_Guide.asciidoc [hunk 36]
```diff
@@ -2693,61 +2589,66 @@ And can be defined using different criteria like:
...
 (here $user_role will be replaced by the real user role of the device and ${switch._portalURL} will be replaced
 by the value of _portalURL defined in the switch config):
 
-  [1:etherneteap&!security_event]
-  merge_answer = yes
-  scope = returnRadiusAccessAccept
-  answer1 = Cisco-AVPair => url-redirect-acl=$user_role;url-redirect=${switch._portalURL}/cep$session_id
+  [Ethernet-EAP-Accept]
+  status=enabled
+  description=Access Accept with VSA when the connection is Ethernet-EAP and when there is no security event
```
Comment (extrafu) [inline docs/PacketFence_Installation_Guide.asciidoc]: Access Accept -> Return Access-Accept

File: docs/PacketFence_Installation_Guide.asciidoc [hunk 37]
```diff
@@ -2693,61 +2589,66 @@ And can be defined using different criteria like:
...
-  #Terminate-Action to 0 means Terminate session
-  #Terminate-Action to 1 means Reauthenticate
-  answer2 = Termination-Action => 1
-  answer3 = Idle-Timeout => 240
+  [Session-timeout Idle-Timeout and terminate action]
+  answer.1=reply:Idle-Timeout = 240
+  status=enabled
+  answer.0=reply:Session-Timeout = 10620..12600
+  top_op=and
+  description=Change the session timeout and
```
Comment (extrafu) [inline docs/PacketFence_Installation_Guide.asciidoc]: and what?

File: docs/PacketFence_Installation_Guide.asciidoc [hunk 38]
```diff
@@ -2693,61 +2589,66 @@ And can be defined using different criteria like:
...
+  status=enabled
+  answer.0=reply:Cisco-AVPair = ip:inacl#101=permit ip any any
+  top_op=and
+  description=Detect ipad
+  scopes=returnRadiusAccessAccept
+  radius_status=RLM_MODULE_OK
+  merge_answer=yes
+  condition=contains(fingerbank_info.device_hierarchy_names, "Apple iPad")
+
+preProcess scope can be usefull when you want to create internal radius attributes that will be used internally, like in the authentication rules.
```
Comment (extrafu) [inline docs/PacketFence_Installation_Guide.asciidoc]: usefull -> useful
radius -> RADIUS

File: docs/PacketFence_Installation_Guide.asciidoc [hunk 39]
```diff
@@ -2693,61 +2589,66 @@ And can be defined using different criteria like:
...
+
+per example we want to add the attribute TLS-Stripped-UserName by extracting the first portion of the TLS-Client-Cert-Common-Name attribute value:
```
Comment (extrafu) [inline docs/PacketFence_Installation_Guide.asciidoc]: per example we -> For example if we

File: docs/PacketFence_Installation_Guide.asciidoc [hunk 40]
```diff
@@ -2693,61 +2589,66 @@ And can be defined using different criteria like:
...
+
+  [eap-tls-preProcess]
+  status=enabled
+  top_op=and
+  condition=radius_request.User-Name =~ "acme.com$"
+  answer.0=TLS-Stripped-UserName = ${BuildFromMatch($radius_request.TLS-Client-Cert-Common-Name,"^[^@]+","$0")}
+  description=preProcess attribute for eap tls
+  scopes=preProcess
+
+This rule will add the TLS-Stripped-UserName radius attribute in the request and you will be able to use it in the authentication/administrations rules.
```
Comment (extrafu) [inline docs/PacketFence_Installation_Guide.asciidoc]: radius -> RADIUS

File: docs/PacketFence_Installation_Guide.asciidoc [hunk 41]
```diff
@@ -3002,6 +2903,28 @@ Click on save and restart radius (pfcmd service radiusd restart).
 
 Now when a device connect with the username bob@acme.com then the authentication and accounting requests will be forwarded to one of the ACME RADIUS servers.
 
+RADIUS EAP Profiles
+~~~~~~~~~~~~~~~~~~~
+
+RADIUS EAP Profiles allow you to select a specific eap profile in FreeRADIUS based on the Realm of the user.
```
Comment (extrafu) [inline docs/PacketFence_Installation_Guide.asciidoc]: eap -> EAP
FreeRADIUS -> same comment as above
Realm -> realm

File: docs/PacketFence_Installation_Guide.asciidoc [hunk 42]
```diff
@@ -3002,6 +2903,28 @@ Click on save and restart radius (pfcmd service radiusd restart).
...
+
+In this eap profile you can define:
```
Comment (extrafu) [inline docs/PacketFence_Installation_Guide.asciidoc]: eap -> EAP

File: docs/PacketFence_Installation_Guide.asciidoc [hunk 43]
```diff
@@ -3002,6 +2903,28 @@ Click on save and restart radius (pfcmd service radiusd restart).
...
+ Certificates configuration.
+ OCSP configuration
+ EAP-Fast configuration
+ TLS Configuration
+
+And link all these configuration together.
+
+Per example the Realm ACME.COM needs to use the CA certificate from ACME CA and the other Realm needs to use the default one.
```
Comment (extrafu) [inline docs/PacketFence_Installation_Guide.asciidoc]: Per example -> For example
Realm -> realm
the other Realm needs -> the other realms need

File: docs/PacketFence_Installation_Guide.asciidoc [hunk 44]
```diff
@@ -3002,6 +2903,28 @@ Click on save and restart radius (pfcmd service radiusd restart).
...
+
+To do that go in 'Configuration -> System Configuration -> RADIUS -> SSL Certificates' and create a new profile.
+Next go in 'Configuration -> System Configuration -> RADIUS -> TLS Profiles' and create a new TLS profile and select the Certificate profile created just before.
+Then create the EAP profile in 'Configuration -> System Configuration -> RADIUS -> EAP Profiles' and create a new EAP profile and select the TLS profile created before (PEAP Profile per exemple)
+
+The last thing to do is to link the EAP profile with your Realm configuration, to achieve that go in 'Configuration -> Policies and Access Control -> Domains -> Realms' and edit the ACME.COM realm (create it if it's not already the case) then choose the EAP profile you created before in the EAP configuration parameter.
```
Comment (extrafu) [inline docs/PacketFence_Installation_Guide.asciidoc]: Realm -> realm

File: docs/PacketFence_Installation_Guide.asciidoc [hunk 45]
```diff
@@ -3002,6 +2903,28 @@ Click on save and restart radius (pfcmd service radiusd restart).
...
+
+Restart radiusd to generate the new RADIUS configuration.
```
Comment (extrafu) [inline docs/PacketFence_Installation_Guide.asciidoc]: radiusd or packetfence-radiusd?

**Comment by fdurand** [comment]
Changes done

File: docs/PacketFence_Installation_Guide.asciidoc (hunk 20 above)
Comment (extrafu) [reply docs/PacketFence_Installation_Guide.asciidoc]: Missing that one

**Comment by fdurand** [comment]
Ready now

File: conf/radius_filters.conf.defaults [hunk 46]
```diff
@@ -0,0 +1,37 @@
+[EXAMPLE_Ethernet-EAP-Accept]
+status=disabled
+description=Return Access Accept with VSA when the connection is Ethernet-EAP and when there is no security event
```
Comment (extrafu) [inline conf/radius_filters.conf.defaults]: Access Accept -> Access-Accept

File: conf/radius_filters.conf.defaults [hunk 47]
```diff
@@ -0,0 +1,37 @@
...
+merge_answer=yes
+condition=not_defined(security.id, "") && contains(connection_type, "Ethernet-EAP")
+answer.0=reply:Cisco-AVPair = url-redirect-acl=$user_role;url-redirect=${switch._portalURL}/cep$session_id
+
+[EXAMPLE_Session-timeout_Idle-Timeout_Terminate_action]
+answer.1=reply:Idle-Timeout = 240
+status=disabled
+answer.0=reply:Session-Timeout = 10620..12600
+top_op=and
+description=Change the session timeout , idle timeout and terminate action
```
Comment (extrafu) [inline conf/radius_filters.conf.defaults]: timeout , -> timeout,

File: conf/radius_filters.conf.defaults [hunk 48]
```diff
@@ -0,0 +1,37 @@
...
+scopes=returnRadiusAccessAccept
+merge_answer=no
+answer.2=reply:Termination-Action = RADIUS-Request
+condition=switch._ip =~ "^(172\\.24\\.237\\.112|172\\.24\\.237\\.114|172\\.24\\.237\\.128)$"
+
+[EXAMPLE_ipad_by_name]
+status=disabled
+answer.0=reply:Cisco-AVPair = ip:inacl#101=permit ip any any
+top_op=and
+description=Detect ipad
```
Comment (extrafu) [inline conf/radius_filters.conf.defaults]: ipad -> Apple iPad

File: conf/radius_filters.conf.defaults [hunk 49]
```diff
@@ -0,0 +1,37 @@
...
+radius_status=RLM_MODULE_OK
+merge_answer=yes
+condition=contains(fingerbank_info.device_hierarchy_names, "Apple iPad")
+
+[EXAMPLE_eap-tls-preProcess]
+status=disabled
+top_op=and
+condition=radius_request.User-Name =~ "acme.com$"
+answer.0=TLS-Stripped-UserName = ${BuildFromMatch($radius_request.TLS-Client-Cert-Common-Name,"^[^@]+","$0")}
+description=preProcess attribute for eap tls
```
Comment (extrafu) [inline conf/radius_filters.conf.defaults]: preProcess attribute for eap tls -> Preprocess attribute for EAP-TLS

File: docs/PacketFence_Installation_Guide.asciidoc [hunk 50]
```diff
@@ -2586,70 +2586,19 @@ And can be defined using different criteria like:
...
-  [ipad_by_name]
-  status=enabled
-  answer.0=reply:Cisco-AVPair = ip:inacl#101=permit ip any any
-  top_op=and
-  description=Detect ipad
-  scopes=returnRadiusAccessAccept
-  radius_status=RLM_MODULE_OK
-  merge_answer=yes
-  condition=contains(fingerbank_info.device_hierarchy_names, "Apple iPad")
+.EXAMPLE_Ethernet-EAP-Accept will return Access Accept (with Cisco-AVPair attribute) when the connection is Ethernet-EAP and when there is no security event.
```
Comment (extrafu) [inline docs/PacketFence_Installation_Guide.asciidoc]: Access Accept -> Access-Accept

**Pull request merged**